extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/tracing/vertx/VertxTracerFilter.java

```diff
@@ -0,0 +1,129 @@
+package io.quarkus.opentelemetry.tracing.vertx;
+
+import java.util.Iterator;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.quarkus.opentelemetry.QuarkusContextStorage;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.ext.web.RoutingContext;
+
+public class VertxTracerFilter implements Handler<RoutingContext> {
+    private static final TextMapPropagator TEXT_MAP_PROPAGATOR = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        // Needed because CDI Request Scope not active yet
+        QuarkusContextStorage.INSTANCE.setRoutingContext(routingContext);
+
+        Instance<Tracer> tracerInstance = CDI.current().select(Tracer.class);
+
+        // Don't do anything if there is no Tracer
+        if (tracerInstance.isResolvable()) {
+            Tracer tracer = tracerInstance.get();
+
```

Comment:
Similar: Can this be injected or have a cached lookup, rather than happening per request?